### PR TITLE
Freebsd fixes

### PR DESCRIPTION
**Pull request by martymac**
Here are the patches we use on FreeBSD for the port of Dolphin, now updated to 5.0-rc
(see : http://www.freshports.org/emulators/dolphin-emu)

They are based on 5.0-rc and will probably need a bit of work to apply on master.

Ganael.

---
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3190)

### CODE REVIEW

**Comment by endrift** [comment]
Where did these translation changes come from? This doesn't seem like the right PR for them.

Anyway, glad to see that FreeBSD is trying to upstream their changes :) I was wondering what would happen to the FreeBSD port if my earlier fixes hadn't gone in.

File: CMakeTests/FindLibUSB.cmake [hunk 1]
```diff
@@ -23,7 +23,13 @@ elseif (NOT LIBUSB_FOUND)
...
     )
 
-    find_library(LIBUSB_LIBRARIES NAMES usb-1.0
+    if(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
+        set(LIBUSB_LIBRARY "usb")
+    else()
+        set(LIBUSB_LIBRARY "usb-1.0")
+    endif()
+
+    find_library(LIBUSB_LIBRARIES NAMES ${LIBUSB_LIBRARY}
```
Comment (endrift) [inline CMakeTests/FindLibUSB.cmake]: Can't you just do `NAMES usb-1.0 usb`?

**Comment by endrift** [comment]
I think some of these fixes may have been present in #3145, which was recently merged into master. It'd be cool if those + some of these could get merged into stable to reduce the number of patches needed downstream.

**Comment by martymac** [comment]
Hi endrift,

I don't know where those translation updates come from ; I've probably made a mistake somewhere when generating the pull request :/

Anyway, I've updated the patches (and also committed them to the ports tree). I hope they will be merged back to master :)

Thanks for your help !
Best regards,
Ganael.

**Comment by JosJuice** [comment]
PRs that target master need to be based on master, otherwise things like the translation updates will be pulled along with it. I suggest rebasing your commits.

**Comment by Fallcrest** [comment]
@martymac Could this be rebased? Someone is doing several other fixes for BSD in PRs [4017](https://github.com/dolphin-emu/dolphin/pull/4017), [4018](https://github.com/dolphin-emu/dolphin/pull/4018), [4019](https://github.com/dolphin-emu/dolphin/pull/4019), [4020](https://github.com/dolphin-emu/dolphin/pull/4020), and [4021](https://github.com/dolphin-emu/dolphin/pull/4021), in case you were interested in weighing in on those.

**Comment by martymac** [comment]
@Fallcrest Unfortunately, those patches are quite old now. I will have to update the port and its patches to 5.0. I'll work on that ASAP (ENOTIME at the moment) and try to post an up-to-date merge request. Also, thanks for the pointers to the ongoing work on FreeBSD support, I'll have a look at that :)

Meanwhile, if you are interested in a clean vision of current patches for version 5.0rc, you can check the port's patches here :

http://svnweb.freebsd.org/ports/head/emulators/dolphin-emu/files/

Best regards,
Ganael

**Comment by RisingFog** [comment]
Is this PR still valid? Or will a new PR be created once it's updated to match current version?

**Comment by martymac** [comment]
Yep, this PR can be closed, as it is quite old. I am still missing time to work on the Dolphin port update right now. I'll open a new PR once it is done (meanwhile, any help is welcome if someone is willing to update the port). Thanks!

**Pull request closed without merging**